www/m/js/widgets/panelSettings.js

```diff
@@ -0,0 +1,244 @@
+			switch (p) {
+				case 'S':
+					r2.cmd('e asm.bytes=false');
+					r2.cmd('e asm.lines=false');
+					r2.cmd('e asm.cmtright=false');
+					break;
+				case 'M':
+					r2.cmd('e asm.bytes=false');
+					r2.cmd('e asm.lines=true');
+					r2.cmd('e asm.lineswidth=8');
+					r2.cmd('e asm.cmtright=false');
+					break;
+				case 'L':
+					r2.cmd('e asm.bytes=true');
+					r2.cmd('e asm.lines=true');
+					r2.cmd('e asm.lineswidth=12');
+					r2.cmd('e asm.cmtright=true');
+					break;
+			};
+		}
+	},
+	decoding: { name: 'decoding', defVal: 'Pseudo', apply: function(p) {
+			switch (p) {
+				case 'Pseudo':
+					r2.cmd('e asm.pseudo=1');
+					r2.cmd('e asm.syntax=intel');
+					break;
+				case 'Opcodes':
+					r2.cmd('e asm.pseudo=0');
+					r2.cmd('e asm.syntax=intel');
+					break;
+				case 'ATT':
+					r2.cmd('e asm.pseudo=0');
+					r2.cmd('e asm.syntax=att');
+					break;
+			};
+		}
+	},
+	mode: { name: 'mode', defVal: 'PA', apply: function(p) {
+			switch (p) {
+				case 'PA':
+					r2.cmd('e io.va=false');
+					break;
+				case 'VA':
+					r2.cmd('e io.va=true');
+					break;
+				case 'Debug':
+					r2.cmd('e io.va=true');
+					r2.cmd('e io.debug=true');
+					break;
+			};
+		}
+	},
+	analHasNext: { name: 'analHasNext', defVal: true, apply: function(p) { console.log('analHasNext is ' + p); } },
+	analSkipNops: { name: 'analSkipNops', defVal: true, apply: function(p) { console.log('analSkipNops is ' + p); } },
+	analNonCode: { name: 'analNonCode', defVal: false, apply: function(p) { console.log('analNonCode is ' + p); } },
+	colors: { name: 'colors', defVal: true, apply: function(p) { inColor = p; } },
+	theme: { name: 'theme', defVal: 'none', apply: function(p) { r2.cmd('eco ' + p); } } // TODO
+};
+
+function panelSettings() {
+	var widget = widgetContainer.getWidget('Settings');
+	var c = widgetContainer.getWidgetDOMWrapper(widget);
+	c.innerHTML = '';
+	updates.registerMethod(widget.getOffset(), panelSettings);
+
+	var grid = document.createElement('div');
+	grid.className = 'mdl-grid';
+	c.appendChild(grid);
+
+	var platform = createGrid(grid, 'Platform');
+	drawPlatform(platform);
+
+	var disassembly = createGrid(grid, 'Disassembly');
+	drawDisassembly(disassembly);
+
+	var coreio = createGrid(grid, 'Core/IO');
+	drawCoreIO(coreio);
+
+	var analysis = createGrid(grid, 'Analysis');
+	drawAnalysis(analysis);
+
+	var colors = createGrid(grid, 'Colors');
+	drawColors(colors);
+
+	var reset = createGrid(grid, 'Reset configuration');
+	uiActionButton(reset, function() {
+		resetConf();
+		update();
+	}, 'RESET');
+
+	componentHandler.upgradeDom();
+}
+
+function savedFromList(list, name, defaultOffset) {
+	var value = defaultOffset;
+	var saved = localStorage.getItem(name);
+	if (saved !== null) {
+		value = list.indexOf(saved);
+	}
+	return value;
+}
+
+function drawPlatform(dom) {
+	var archs = ['x86', 'arm', 'mips', 'java', 'dalvik', '6502', '8051', 'h8300', 'hppa', 'i4004', 'i8008', 'lh5801',
```

Comment:
agree. you can grab this list from r2.cmdj("e asm.arch=?j") (this will return a json object) if you want only the name just use ?q instead of ?j